Source/WebCore/platform/SimpleCaretAnimator.cpp

```diff
@@ -40,13 +40,13 @@ void SimpleCaretAnimator::updateAnimationProperties(ReducedResolutionSeconds cur
     auto caretBlinkInterval = RenderTheme::singleton().caretBlinkInterval();
 
     // Ensure the caret is always visible when blinking is suspended.
-    if (isBlinkingSuspended() && isVisible()) {
+    if (isBlinkingSuspended() && m_presentationProperties.blinkState == PresentationProperties::BlinkState::On) {
         m_blinkTimer.startOneShot(caretBlinkInterval);
         return;
     }
 
     if (currentTime - m_lastTimeCaretPaintWasToggled >= caretBlinkInterval) {
-        setVisible(m_blinkState == CaretBlinkState::Off);
+        setBlinkState(m_presentationProperties.blinkState == PresentationProperties::BlinkState::On ? PresentationProperties::BlinkState::Off : PresentationProperties::BlinkState::On);
```

Comment:
This is a bit unwieldy

Reply:
ya I agree, not sure of a better way though. any ideas?